### PR TITLE
HEEDLS-814 Fix bugs with PRN messaging and input filling

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-814

### Description
Fixed the bug with form inputs not getting filled when returning to the admin learner information page from the summary and the wording so that it is correct for You/They for where the admins are editing/registering vs self registration/edit.

### Screenshots
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Views/Shared/_EditRegistrationNumber.cshtml [hunk 1]
```diff
@@ -15,18 +14,19 @@
...
   <fieldset id="HasProfessionalRegistrationNumber" class="nhsuk-fieldset" aria-describedby="professional-registration-hint">
 
     <legend class="nhsuk-fieldset__legend nhsuk-fieldset__legend--l">
       <div class="nhsuk-label">
-        Do @(isEditDelegateByAdmin ? "they" : "you") have a Professional Registration Number?
+        Do @(!Model.IsSelfRegistrationOrEdit ? "they" : "you") have a Professional Registration Number?
       </div>
       <div class="nhsuk-hint nhsuk-u-margin-bottom-0" id="professional-registration-hint">
-        @(isEditDelegateByAdmin ? "They" : "You") should have a professional registration number if @(isEditDelegateByAdmin ? "they" : "you") are a health professional registered with a professional body such as the NMC, GMC or GDC.
+        @(!Model.IsSelfRegistrationOrEdit ? "They" : "You") should have a professional registration number if @(!Model.IsSelfRegistrationOrEdit ? "they" : "you") are a health professional registered with a professional body such as the NMC, GMC or GDC.
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web/Views/Shared/_EditRegistrationNumber.cshtml:29]: Is it worth making these a variable to neaten up the string a bit? E.g. 
`var subject = Model.IsSelfRegistrationOrEdit ? "you" : "they";`
`var capitalisedSubject = Model.IsSelfRegistrationOrEdit? "You" : "They";`

File: DigitalLearningSolutions.Web/ViewModels/Register/LearnerInformationViewModel.cs [hunk 2]
```diff
@@ -10,12 +10,18 @@ public class LearnerInformationViewModel : IEditProfessionalRegistrationNumbers
     {
         public LearnerInformationViewModel() { }
 
-        public LearnerInformationViewModel(RegistrationData data)
+        public LearnerInformationViewModel(RegistrationData data, bool isSelfRegistration)
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web/ViewModels/Register/LearnerInformationViewModel.cs:13]: Should this param be renamed to include the "OrEdit" of the property?

File: DigitalLearningSolutions.Web/Views/Shared/_EditRegistrationNumber.cshtml [hunk 3]
```diff
@@ -15,18 +14,19 @@
...
         @(isRegisterJourney ? "If you don't have it to hand, you can provide it later by visiting My account / Edit details." : "")
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web/Views/Shared/_EditRegistrationNumber.cshtml:30]: I think this should read "If you don't have it to hand, **they** can provide it later by visiting My account / Edit details." as the admin won't be able to change the value for this account on their own My account page.

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Minor wording change I think can be done, but looks good.

File: DigitalLearningSolutions.Web/Views/Shared/_EditRegistrationNumber.cshtml [hunk 4]
```diff
@@ -15,18 +14,19 @@
   const string professionalRegConditionalId = "professional-reg-conditional-Id";
 
   var isRegisterJourney = Model is LearnerInformationViewModel;
-  var isEditDelegateByAdmin = Model is EditDelegateViewModel;
 }
 
+<input type="hidden"asp-for="IsSelfRegistrationOrEdit"/>
```
Comment (GeorgeBlackwell) [inline DigitalLearningSolutions.Web/Views/Shared/_EditRegistrationNumber.cshtml:19]: Should probably be a space between these

**Review by GeorgeBlackwell — COMMENTED** [review]
One very minor comment, nice work 🙂

File: DigitalLearningSolutions.Web/ViewModels/Register/LearnerInformationViewModel.cs (hunk 2 above)
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/ViewModels/Register/LearnerInformationViewModel.cs:13]: I don't think so, this parameter only applies to a model being used for Registration journeys.

File: DigitalLearningSolutions.Web/Views/Shared/_EditRegistrationNumber.cshtml (hunk 3 above)
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/Views/Shared/_EditRegistrationNumber.cshtml:30]: I think this actually needs the middle ground using the subject variable we've just created, although an admin could edit it in the Edit delegate page.

Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/Views/Shared/_EditRegistrationNumber.cshtml:30]: Changed to $"If you don't have it to hand, {subject} can provide it later by visiting My account / Edit details."

**Review by GeorgeBlackwell — APPROVED** [review]

**Pull request merged**